inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp

```diff
@@ -311,6 +312,10 @@ CNNNetworkNGraphImpl::reshape(const std::map<std::string, std::vector<size_t>>&
     try {
         auto params = _ngraph_function->get_parameters();
 
+        ngraph::pass::Manager ssr_manager;
+        ssr_manager.register_pass<ngraph::pass::SmartReshape>();
+        ssr_manager.run_passes(_ngraph_function);
```

Comment:
If you apply this transformation pass to the original nGraph function looks like you can do it only one time in the constructor.

Reply:
Done that. But it is weird to perform too much there. But I agree -- we should do it once.